Can't build kodi on Ubuntu 24.04.2 LTS

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report

Can't finish build on Ubuntu 24.04.2 LTS

### Describe the bug
Here is a clear and concise description of what the problem is:

I'm new to building software from source but because there is now no Ubuntu PPA, I tried to build by following the [ubuntu build instructions](https://github.com/xbmc/xbmc/blob/master/docs/README.Ubuntu.md) and then the [general linux build guide](https://github.com/xbmc/xbmc/blob/master/docs/README.Linux.md), and it almost finishes (gets to [100%] Built target legacy_interface), but it gets some kind of mysql error at the end.

I have done a `git pull` in `~/kodi` folder and it says "already up to date".

<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

The build is successful and everyone is happy.
<!--- Tell us what should happen -->
<!--- Put your text below this line -->



## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->

`cmake ../kodi -DENABLE_INTERNAL_FFMPEG=ON -DCMAKE_INSTALL_PREFIX=/usr/local -DCORE_PLATFORM_NAME=x11 -DAPP_RENDER_SYSTEM=gl`

`cmake --build . -- VERBOSE=1 -j$(getconf _NPROCESSORS_ONLN)`

```
<snipped ~2300 lines>

[100%] Built target legacy_interface
In file included from /usr/include/mysql/mysql.h:84,
                 from /home/redacted/kodi/xbmc/dbwrappers/mysqldataset.cpp:28:
/usr/include/mysql/client_plugin.h:116:8: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  116 | struct MYSQL;
      |        ^~~~~
In file included from /home/redacted/kodi/xbmc/dbwrappers/mysqldataset.cpp:9:
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:123:58: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  123 |   int (*authenticate_user)(MYSQL_PLUGIN_VIO *vio, struct MYSQL *mysql);
      |                                                          ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:125:65: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  125 |                                                          struct MYSQL *mysql,
      |                                                                 ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:147:57: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  147 | struct st_mysql_client_plugin *mysql_load_plugin(struct MYSQL *mysql,
      |                                                         ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:167:59: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  167 | struct st_mysql_client_plugin *mysql_load_plugin_v(struct MYSQL *mysql,
      |                                                           ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:181:64: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  181 | struct st_mysql_client_plugin *mysql_client_find_plugin(struct MYSQL *mysql,
      |                                                                ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/client_plugin.h:200:12: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  200 |     struct MYSQL *mysql, struct st_mysql_client_plugin *plugin);
      |            ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/mysql.h:299:16: error: using typedef-name ‘using MYSQL = struct st_mysql’ after ‘struct’
  299 | typedef struct MYSQL {
      |                ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: ‘using MYSQL = struct st_mysql’ has a previous declaration here
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/usr/include/mysql/mysql.h:337:3: error: conflicting declaration ‘typedef int MYSQL’
  337 | } MYSQL;
      |   ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.h:16:7: note: previous declaration as ‘using MYSQL = struct st_mysql’
   16 | using MYSQL = st_mysql;
      |       ^~~~~
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.cpp: In member function ‘virtual int dbiplus::MysqlDatabase::connect(bool)’:
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.cpp:183:20: warning: ‘bool mysql_ssl_set(MYSQL*, const char*, const char*, const char*, const char*, const char*)’ is deprecate
d: Use mysql_options() instead. [-Wdeprecated-declarations]
  183 |       mysql_ssl_set(conn, key.empty() ? nullptr : key.c_str(),
      |       ~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  184 |                     cert.empty() ? nullptr : cert.c_str(), ca.empty() ? nullptr : ca.c_str(),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  185 |                     capath.empty() ? nullptr : capath.c_str(),
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  186 |                     ciphers.empty() ? nullptr : ciphers.c_str());
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/mysql/mysql.h:465:1: note: declared here
  465 | mysql_ssl_set(MYSQL *mysql, const char *key, const char *cert, const char *ca,
      | ^~~~~~~~~~~~~
gmake[2]: *** [build/dbwrappers/CMakeFiles/dbwrappers.dir/build.make:146: build/dbwrappers/CMakeFiles/dbwrappers.dir/mysqldataset.cpp.o] Error 1
gmake[2]: Leaving directory '/home/redacted/kodi-build'
gmake[1]: *** [CMakeFiles/Makefile2:9802: build/dbwrappers/CMakeFiles/dbwrappers.dir/all] Error 2
gmake[1]: Leaving directory '/home/redacted/kodi-build'
gmake: *** [Makefile:146: all] Error 2
```

### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
Ubuntu 24.04.2 LTS
 - Kodi version:
git nightly

Thanks for the reply but unfortunately that didn't work.

`sudo apt remove libmysqlclient-dev`
`cmake --build . -- VERBOSE=1 -j$(getconf _NPROCESSORS_ONLN)`
```
<snip>
[100%] Built target legacy_interface
/home/redacted/kodi/xbmc/dbwrappers/mysqldataset.cpp:27:10: fatal error: mysql/errmsg.h: No such file or directory
   27 | #include <mysql/errmsg.h>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [build/dbwrappers/CMakeFiles/dbwrappers.dir/build.make:146: build/dbwrappers/CMakeFiles/dbwrappers.dir/mysqldataset.cpp.o] Error 1
gmake[2]: Leaving directory '/home/redacted/kodi-build'
gmake[1]: *** [CMakeFiles/Makefile2:9802: build/dbwrappers/CMakeFiles/dbwrappers.dir/all] Error 2
gmake[1]: Leaving directory '/home/redacted/kodi-build'
gmake: *** [Makefile:146: all] Error 2

```

--- a/xbmc/dbwrappers/mysqldataset.cpp
+++ b/xbmc/dbwrappers/mysqldataset.cpp
@@ -25,10 +25,8 @@
 
 #ifdef HAS_MYSQL
 #include <mysql/errmsg.h>
-#include <mysql/mysql.h>
 #elif defined(HAS_MARIADB)
 #include <mariadb/errmsg.h>
-#include <mariadb/mysql.h>
 #endif
 
 #ifdef TARGET_POSIX

--- a/xbmc/dbwrappers/mysqldataset.h
+++ b/xbmc/dbwrappers/mysqldataset.h
@@ -12,8 +12,11 @@
 
 #include <string>
 
-struct st_mysql;
-using MYSQL = st_mysql;
+#ifdef HAS_MYSQL
+#include <mysql/mysql.h>
+#elif defined(HAS_MARIADB)
+#include <mariadb/mysql.h>
+#endif
 
 namespace dbiplus
 {

